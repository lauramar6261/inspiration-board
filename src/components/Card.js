import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  render() {
    let icon = this.props.emoji;
    if (icon !== undefined) {
      icon = emoji.getUnicode(icon);
    }
    console.log(icon)
    return (
      <div className="card">
        <div className="card__content">
        {this.props.text}
        <section className="card__content-text">

        </section>
        <section className="card__content-emoji">
        {icon}
        </section>
        <button
          onClick={() => this.props.deleteCardCallback(this.props.id)}
          type="button"
          className="card__delete"
          aria-label="Close"
        >
          <span aria-hidden="true">&times;</span>
        </button>
        </div>
      </div>
    )
  }
}

Card.propTypes = {

};

export default Card;
