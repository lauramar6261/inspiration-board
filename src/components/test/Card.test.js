import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {

  it('will match the Card Snapshot', () => {
    const wrapper = shallow( <Card
      text="hello test"
      emoji=":)"
      id={500}
    /> );

    expect(wrapper).toMatchSnapshot();
  });
});
