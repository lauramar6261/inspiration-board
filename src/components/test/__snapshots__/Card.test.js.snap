// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card will match the Card Snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    hello test
    <section
      className="card__content-text"
    />
    <section
      className="card__content-emoji"
    />
    <button
      aria-label="Close"
      className="card__delete"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
    </button>
  </div>
</div>
`;
